inc/classes/class-srm-redirect.php

```diff
@@ -116,13 +138,18 @@ public function match_redirect( $requested_path ) {
 
 			// check if requested path is the same as the redirect from path
 			if ( $enable_regex ) {
-				$matched_path = preg_match( '@' . $redirect_from . '@' . $regex_flag, $requested_path );
+				$match_query_params = false;
+				$matched_path       = preg_match( '@' . $redirect_from . '@' . $regex_flag, $requested_path );
 			} else {
 				if ( $case_insensitive ) {
 					$redirect_from = strtolower( $redirect_from );
 				}
 
-				$matched_path = ( $normalized_requested_path === $redirect_from );
+				// only compare query params if the $redirect_from value contains parameters
+				$match_query_params = apply_filters( 'srm_match_query_params', strpos( $redirect_from, '?' ) );
+
+				$to_match     = ( ! $match_query_params && ! empty( $normalized_requested_path_no_query ) ) ? $normalized_requested_path_no_query : $normalized_requested_path;
```

Comment:
This line can be simplified to
```suggestion
				$to_match     = $match_query_params ? $normalized_requested_path : untrailingslashit( stripslashes( $parsed_requested_path['path'] ) );
```

Reply:
This change causes tests to fail, as untrailingslashit( stripslashes( $parsed_requested_path['path'] ) ) will return an empty string for redirects from the site root.